scrapy/contracts/__init__.py

```diff
@@ -51,8 +51,11 @@ def from_method(self, method, results):
             # calculate request args
             args, kwargs = get_spec(Request.__init__)
             kwargs['callback'] = method
+            ignored_errors = []
             for contract in contracts:
                 kwargs = contract.adjust_request_args(kwargs)
+                if contract.name == 'ignore':
```

Comment:
Actually the ```@ignore``` contract is not a "real" contract in sense it does not make assertion on response or spider output, it modify the ContractManager behaviour defining which errors might be ignored. 

So I think it doesn't make sense to add a method to define which errors should be ignore in the others contracts because the errors are raised before the contracts are run (mainly by the downloader).

I agree in term of design this is not ideal but it should works  to handle errors occurring when websites are unavailable.